src/extension-consumer/src/produce/mod.rs

```diff
@@ -1,149 +1,269 @@
 use std::fs::File;
 use std::io::{BufReader, BufRead};
-use std::path::PathBuf;
+use std::path::{PathBuf, Path};
 use structopt::StructOpt;
+use tracing::debug;
 
 use fluvio::{Fluvio, TopicProducer};
 use fluvio_types::print_cli_ok;
 use crate::common::FluvioExtensionMetadata;
-use crate::Result;
-
-/// Produce log configuration parameters
-#[derive(Debug)]
-pub struct ProduceLogConfig {
-    pub topic: String,
-    pub partition: i32,
-    pub continuous: bool,
-}
-
-#[derive(Debug)]
-pub enum FileRecord {
-    Lines(PathBuf),
-    Files(Vec<PathBuf>),
-}
+use crate::{Result, ConsumerError};
 
 // -----------------------------------
 // CLI Options
 // -----------------------------------
 
+/// Write messages to a topic/partition
+///
+/// When no '--files' are provided, the producer will read from 'stdin'
+/// and send each line of input as one record.
+///
+/// If one or more files are given with '--files', each file is sent as
+/// one entire record.
+///
+/// If '--key-separator' or '--json-path' are used, records are sent as
+/// key/value pairs, and the keys are used to determine which partition
+/// the records are sent to.
 #[derive(Debug, StructOpt)]
 pub struct ProduceLogOpt {
     /// The name of the Topic to produce to
     #[structopt(value_name = "topic")]
     pub topic: String,
 
-    /// The ID of the Partition to produce to
-    #[structopt(
-        short = "p",
-        long = "partition",
-        value_name = "integer",
-        default_value = "0"
-    )]
-    pub partition: i32,
-
-    /// Send messages in an infinite loop
-    #[structopt(short = "C", long = "continuous")]
-    pub continuous: bool,
-
-    /// Send each line of the file as its own Record
-    #[structopt(
-        short = "l",
-        long = "record-per-line",
-        value_name = "filename",
-        parse(from_os_str)
-    )]
-    record_per_line: Option<PathBuf>,
-
-    /// Send an entire file as a single Record
-    #[structopt(
-        short = "r",
-        long = "record-file",
-        value_name = "filename",
-        parse(from_os_str),
-        conflicts_with = "record-per-line"
-    )]
-    record_file: Vec<PathBuf>,
+    /// Send each line of input as its own record (using '\n')
+    #[structopt(hidden = true, short, long)]
+    pub lines: bool,
+
+    /// Print progress output when sending records
+    #[structopt(short, long)]
+    pub verbose: bool,
+
+    /// Sends key/value records split on the first instance of the separator.
+    #[structopt(long, validator = validate_key_separator)]
+    pub key_separator: Option<String>,
+
+    /// Sends key/value JSON records where the key is selected using this JSON path.
+    #[structopt(long, conflicts_with("key-separator"))]
+    pub json_path: Option<String>,
+
+    /// Paths to files to produce to the topic. If absent, producer will read stdin.
+    #[structopt(short, long)]
+    pub files: Vec<PathBuf>,
```

Comment:
I don't think we should handle multiple files which is opinionated process (for example, what is ordering of file input).  Also, we should support short format `-f`

Reply:
Need to re-think about support multiple files whether it's warranted or not.  Let's remove until we got come up with better design